src/NuGet.Services.AzureSearch/SearchService/AzureSearchService.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using NuGet.Services.AzureSearch.Wrappers;
+
+namespace NuGet.Services.AzureSearch.SearchService
+{
+    public class AzureSearchService : ISearchService
+    {
+        private readonly ISearchParametersBuilder _parametersBuilder;
+        private readonly ISearchIndexClientWrapper _searchIndex;
+        private readonly ISearchIndexClientWrapper _hijackIndex;
+        private readonly ISearchResponseBuilder _responseBuilder;
+
+        public AzureSearchService(
+            ISearchParametersBuilder parametersBuilder,
+            ISearchIndexClientWrapper searchIndex,
+            ISearchIndexClientWrapper hijackIndex,
+            ISearchResponseBuilder responseBuilder)
+        {
+            _parametersBuilder = parametersBuilder ?? throw new ArgumentNullException(nameof(parametersBuilder));
+            _searchIndex = searchIndex ?? throw new ArgumentNullException(nameof(searchIndex));
+            _hijackIndex = hijackIndex ?? throw new ArgumentNullException(nameof(hijackIndex));
+            _responseBuilder = responseBuilder ?? throw new ArgumentNullException(nameof(responseBuilder));
+        }
+
+        public async Task<V2SearchResponse> V2SearchAsync(V2SearchRequest request)
+        {
+            if (request.IgnoreFilter)
+            {
+                return await UseHijackIndexAsync(request);
+            }
+            else
+            {
+                return await UseSearchIndexAsync(request);
+            }
+        }
+
+        public async Task<V3SearchResponse> V3SearchAsync(V3SearchRequest request)
+        {
+            var text = _parametersBuilder.GetSearchTextForV3Search(request);
+            var parameters = _parametersBuilder.GetSearchParametersForV3Search(request);
+
+            var stopwatch = Stopwatch.StartNew();
```

Comment:
What do you think about introducing a wrapper, so this `Stopwatch.StartNew()`, `<do your thing>`, `stopwatch.Stop()` pattern is not repeated over and over?
Something like 
```C#
static TimeSpan Measure(Func<Task> action)
{
  var sw = Stopwatch.StartNew();
  action();
  sw.Stop();
  return sw.Elapsed;
}
```
Then in the code:
```C#
var sw = Something.Measure(async () => result = await _searchIndex.Documents.SearchAsync(...));
```

Reply:
Sure!